### PR TITLE
JAMES-1734 As an authenticated JMAP user, I can prentend to be someon…

### PR DESCRIPTION
**Pull request by chibenwa**
…e else in the mails I send

### CODE REVIEW

**Comment by aduprat** [comment]
:+1:

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java [hunk 1]
```diff
@@ -666,7 +666,7 @@ public void setMessageShouldReturnCreatedMessageWhenSendingMessage() {
                 "    \"setMessages\","+
                 "    {" +
                 "      \"create\": { \"" + messageCreationId  + "\" : {" +
-                "        \"from\": { \"name\": \"MAILER-DAEMON\", \"email\": \"postmaster@example.com\"}," +
+                "        \"from\": { \"name\": \"Me\", \"email\": \"" + username + "\"}," +
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: username is not email, you have to change variable name or create a new one.

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java [hunk 2]
```diff
@@ -111,11 +112,13 @@ public SetMessagesResponse process(SetMessagesRequest request, MailboxSession ma
             throw Throwables.propagate(e);
         }
 
+        ImmutableList<String> allowedSenders = ImmutableList.of(mailboxSession.getUser().getUserName());
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java]: the type of the variable is too precise

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java [hunk 3]
```diff
@@ -127,10 +130,10 @@ public SetMessagesResponse process(SetMessagesRequest request, MailboxSession ma
     }
 
     private Map<CreationMessageId, SetError> handleCreationErrors(Predicate<CreationMessage> invalidMessagesTester,
-                                                                  SetMessagesRequest request) {
+                                                                  SetMessagesRequest request, ImmutableList<String> allowedFromMailAddresses) {
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java]: argument type is too precise

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/CreationMessage.java [hunk 4]
```diff
@@ -311,18 +311,32 @@ public ZonedDateTime getDate() {
...
+    public List<ValidationResult> validate(ImmutableList<String> allowedFromMailAddresses) {
         ImmutableList.Builder<ValidationResult> errors = ImmutableList.builder();
         assertValidFromProvided(errors);
         assertAtLeastOneValidRecipient(errors);
         assertSubjectProvided(errors);
+        assertUsedAllowedFromMailAddress(errors, allowedFromMailAddresses);
         return errors.build();
     }
 
+    private void assertUsedAllowedFromMailAddress(ImmutableList.Builder<ValidationResult> errors, ImmutableList<String> allowedFromMailAddresses) {
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/CreationMessage.java]: It's not a validity concern, it's an access control one. It doesn't belong to validate method.

**Comment by chibenwa** [comment]
test this please

**Comment by mbaechler** [comment]
please answer on my comments when you fix things

**Comment by chibenwa** [comment]
As you want

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java (hunk 1 above)
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: Ok

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/CreationMessage.java (hunk 4 above)
Comment (chibenwa) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/CreationMessage.java]: Ok. I moved this concern to the processor.

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java (hunk 3 above)
Comment (chibenwa) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java]: Ok

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java (hunk 2 above)
Comment (chibenwa) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java]: Ok

**Comment by mbaechler** [comment]
:+1:

**Pull request closed without merging**